docs/decisions/0002-enterprise-catalog-ida.rst

```diff
@@ -0,0 +1,64 @@
+
+Our main goals are as follows:
+
+1. Get the response time of the enterprise customer catalog endpoints to be below 1 second for 99% of calls.
+
+2. Since many features do not hit the enterprise catalog api endpoints but perform the same calculations directly in code, we would also like to improve the performance of those code paths and drastically reduce the calls to the discovery service.
+
+What is Enterprise Catalog?
+___________________________
+
+Enterprise catalogs are essentially saved elasticsearch queries attached to an Enterprise Customer. There can be many catalogs configured for a given customer, which enables different sets of learners from the same Enterprise to be given access to different sets of courses, and allows different discounts to be configured for different sets of courses.
+
+We expose two apis that are frequently used by various enterprise features: one to retrieve all of the content metadata that would be returned by the configured catalog query, and one to determine if a given course id or program id is part of the set of content returned by the catalog query. These apis exist both as python apis and REST apis that can be hit from within our edX ecosystem and externally from customers. They are wrappers around the search/all endpoint in the discovery service. Although there is caching in place, the calls to discovery are implemented synchronously, and we've found that half the time requests do not hit the cache, and thus the most of the time is spent waiting on the discovery service to return results.
+
+Who’s Calling What now?
+_______________________
+
+Here is a diagram showing a simplified view of the calls to these apis. The yellow boxes represent the features or user workflows that trigger calls to the various apis. The blue clouds represent REST apis, and the purple rounded boxes represent python code apis. Solid lines are synchronous api calls and dotted lines are code apis (not hitting the network).
+
+.. image:: ../images/enterprise-catalog-api-calls.png
+
+Decisions
+=========
+
+In order to address these performance issues and to more effectively scale and build new features related to enterprise catalogs in the future, we will create a new IDA (Independently Deployable Application) to replace the existing enteprise catalog functionality.
+This IDA will support the existing API contracts established in the edx-enterprise API, but will be a new implementation separate from what we do today.
+
+The microservice will be supported by the following relational data model:
```

Comment:
Content catalog and catalog content seem a little confusingly named

Reply:
agreed, i will probably update the name. nothing is set in stone here so it's not too late to rename!